flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -458,4 +462,11 @@ private static String determineTaskManagerBindAddressByConnectingToResourceManag
 		HostBindPolicy bindPolicy = HostBindPolicy.fromString(configuration.getString(TaskManagerOptions.HOST_BIND_POLICY));
 		return bindPolicy == HostBindPolicy.IP ? taskManagerAddress.getHostAddress() : taskManagerAddress.getHostName();
 	}
+
+	@VisibleForTesting
+	static ResourceID getTaskManagerResourceID(Configuration config) throws Exception {
+		final String resourceID = config.get(TaskManagerOptions.TASK_MANAGER_RESOURCE_ID);
+		return resourceID != null ?
+			new ResourceID(resourceID) : new ResourceID(InetAddress.getLocalHost().getHostName() + "-" + new AbstractID().toString().substring(0, 6));
```

Comment:
I'm wondering whether the address the `RpcService` binds to would be more meaningful. Moreover, we could also add the port of the `RpcService`. But as a safety net we should still add a random suffix to filter out if a TM gets restarted on the same machine with the same port.

Reply:
How to set the `resourceID` when we use a random port for TaskManager rpc? Also the TaskManager could bind to an ip address. So i prefer a random uuid just as before when `TASK_MANAGER_RESOURCE_ID` is not configured.

Reply:
By creating the `ResourceID` after the `RpcService` has been started, one could obtain this information.

Reply:
Well my only concern is whether it is safe to put some special characters(e.g. `:`) in the `ResourceID`. I am not aware of this could happen.

Reply:
I think this should be fine. The {{ResourceID}} is defined to contain a string field.